x-pack/plugins/triggers_actions_ui/public/application/hooks/use_alert_data_view.ts

```diff
@@ -5,72 +5,155 @@
+  const isOnlySecurity =
+    featureIds.length === 1 && (featureIds as AlertConsumers[]).includes(AlertConsumers.SIEM);
 
-  const indexNames = useAsync(async () => {
-    const { index_name: indexNamesStr } = await http.get<{ index_name: string[] }>(
-      `${BASE_RAC_ALERTS_API_PATH}/index`,
-      {
-        query: { features },
-      }
+  const hasSecurityAndO11yFeatureIds =
+    featureIds.length > 1 && (featureIds as AlertConsumers[]).includes(AlertConsumers.SIEM);
+
+  const queryIndexNameFn = () => {
+    return fetchAlertIndexNames({ http, features });
+  };
+
+  const queryAlertFieldsFn = () => {
+    return fetchAlertFields({ http, featureIds });
+  };
+
+  const onErrorFn = () => {
+    toasts.addDanger(
+      i18n.translate('xpack.triggersActionsUI.useAlertDataView.useAlertDataMessage', {
+        defaultMessage: 'Unable to load alert data view',
+      })
     );
+  };
 
-    return indexNamesStr;
-  }, [features]);
+  const {
+    data: indexNames,
+    isSuccess: isIndexNameSuccess,
+    isInitialLoading: isIndexNameInitialLoading,
+    isLoading: isIndexNameLoading,
+  } = useQuery({
+    queryKey: ['loadAlertIndexNames'],
+    queryFn: queryIndexNameFn,
+    onError: onErrorFn,
+    refetchOnWindowFocus: false,
+    enabled: featureIds.length > 0 && !hasSecurityAndO11yFeatureIds,
+  });
 
-  const fields = useAsync(async () => {
-    const { fields: alertFields } = await http.get<{ fields: FieldSpec[] }>(
-      `${BASE_RAC_ALERTS_API_PATH}/browser_fields`,
-      {
-        query: { featureIds },
+  const {
+    data: alertFields,
+    isSuccess: isAlertFieldsSuccess,
+    isInitialLoading: isAlertFieldsInitialLoading,
+    isLoading: isAlertFieldsLoading,
+  } = useQuery({
+    queryKey: ['loadAlertFields'],
+    queryFn: queryAlertFieldsFn,
+    onError: onErrorFn,
+    refetchOnWindowFocus: false,
+    enabled: !isOnlySecurity && !hasSecurityAndO11yFeatureIds,
+  });
+
+  useEffect(() => {
+    return () => {
+      if ((dataview ?? []).length > 0) {
+        dataview?.map(async (dv) => {
+          if (dv.id) {
+            await dataService.dataViews.delete(dv.id);
+          }
+        });
       }
-    );
-    return alertFields;
-  }, [features]);
+    };
+  });
 
-  const dataview = useMemo(
-    () =>
-      !fields.loading &&
-      !indexNames.loading &&
-      fields.error === undefined &&
-      indexNames.error === undefined
-        ? ([
-            {
-              title: (indexNames.value ?? []).join(','),
-              fieldFormatMap: {},
-              fields: (fields.value ?? [])?.map((field) => {
-                return {
-                  ...field,
-                  ...(field.esTypes && field.esTypes.includes('flattened')
-                    ? { type: 'string' }
-                    : {}),
-                };
-              }),
-            },
-          ] as unknown as DataView[])
-        : undefined,
-    [fields, indexNames]
-  );
+  useEffect(() => {
+    async function createDataView() {
```

Comment:
nitpick: do we need this function? can we just inline it in line `101`?

Reply:
you need this function because you are inside of useEffect and it can not be async function